### PR TITLE
P2022 698/create login activity

### PR DESCRIPTION
**Pull request by Lucas-Fresko**
Creates LoginActivity with a header which appears on app launch.

### CODE REVIEW

**Comment by Lucas-Fresko** [comment]
There are 2 minor formatting discrepancies in the manifest file.

File: app/src/main/res/layout/activity_login.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".LoginActivity">
+
+  <LinearLayout
+      android:layout_width="match_parent"
+      android:layout_height="48dp"
```
Comment (taurelas) [inline app/src/main/res/layout/activity_login.xml:12]: could you please extract this value to a "login_header_size" resource ?
you can either 
- try and move the cursor on the "48dp" and press option + enter (or ALT+ENtER on Windows I THINK) and choose extract dimension resource
- or manually create the `dimens.xml` in `res/values` and add the relevant tag

Here are examples and a bit more info: https://developer.android.com/guide/topics/resources/more-resources#Dimension

File: app/src/main/res/layout/activity_login.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      android:layout_marginTop="24dp"
```
Comment (taurelas) [inline app/src/main/res/layout/activity_login.xml:13]: same story here, choose appropriate name and add a resource for this dimension

File: app/src/main/res/layout/activity_login.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      android:background="@android:color/holo_blue_dark"
+      android:orientation="horizontal"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent">
+
+    <TextView
+        android:id="@+id/loginHeaderText"
+        android:layout_width="match_parent"
+        android:layout_height="48dp"
```
Comment (taurelas) [inline app/src/main/res/layout/activity_login.xml:23]: are you trying to say that the height of this textview should MATCH the encompassing layout? If so, there is a better way to do it.

File: app/src/main/res/layout/activity_login.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        android:layout_weight="1"
+        android:gravity="center"
+        android:text="@string/loginHeaderText"
+        android:textColor="@color/black"
+        android:textSize="24sp" />
```
Comment (taurelas) [inline app/src/main/res/layout/activity_login.xml:28]: again, same story with dimens, let's extract this value.

**Review by taurelas — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_login.xml (hunk 3 above)
Comment (Lucas-Fresko) [reply app/src/main/res/layout/activity_login.xml:23]: That wasn't the intent, I've changed it to wrap_content.

File: app/src/main/res/layout/activity_login.xml (hunk 1 above)
Comment (Lucas-Fresko) [reply app/src/main/res/layout/activity_login.xml]: Commit pushed.

**Comment by Lucas-Fresko** [comment]
Is the margin supposed to be applied to header layout? Or would it be better if it applies to text view? I seem to remember mention of a white band during the meeting as the purpose but that doesn't strike me as conventional for a header.

File: app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="header_height_dimension">48dp</dimen>
```
Comment (taurelas) [inline app/src/main/res/values/dimens.xml:3]: let's change it to "header_height", the word dimension is redundant

File: app/src/main/res/values/dimens.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+    <dimen name="header_margin_top_dimension">24dp</dimen>
```
Comment (taurelas) [inline app/src/main/res/values/dimens.xml:4]: again, let's remove the "dimension"

**Comment by taurelas** [comment]
> Is the margin supposed to be applied to header layout? Or would it be better if it applies to text view? I seem to remember mention of a white band during the meeting as the purpose but that doesn't strike me as conventional for a header.

Personally I thought by the "header" we meant a text. You added a background and it became a sort of a toolbar but not directly on top of the screen. I would just leave it as it is for now - the AC's are met and we have a screen that clearly shows it's for login

Normally we should have designs ready which make things clearer.

**Review by taurelas — CHANGES_REQUESTED** [review]

**Review by taurelas — APPROVED** [review]
Well done!

**Pull request merged**